src/TufValidatedComposerRepository.php

```diff
@@ -57,24 +59,40 @@ public function __construct(array $repoConfig, IOInterface $io, Config $config,
             $fs = new Filesystem();
             $fs->ensureDirectoryExists($repoPath);
 
-            // We expect the repository to have a root metadata file in a known
-            // good state. Copy that file to our persistent storage location if
-            // it doesn't already exist.
-            $rootFile = $repoPath . '/root.json';
-            if (!file_exists($rootFile)) {
-                $sourcePath = realpath($repoConfig['tuf']['root']);
-                if (!$fs->copy($sourcePath, $rootFile)) {
-                    throw new FilesystemException("Could not copy '$sourcePath' to '$rootFile");
-                }
-            }
-
-            // For unit testing purposes, allow the updater instance to be passed in
-            // the plugin configuration.
-            if (isset($repoConfig['tuf']['_updater'])) {
-                $this->updater = $repoConfig['tuf']['_updater'];
-            } else {
-                $fetcher = GuzzleFileFetcher::createFromUri($url);
-                $this->updater = new ComposerCompatibleUpdater($fetcher, [], new FileStorage($repoPath));
+            // For unit testing purposes, allow the updater and/or file fetcher instances
+            // to be passed in the repository configuration.
+            $fetcher = $repoConfig['tuf']['_fileFetcher'] ?? GuzzleFileFetcher::createFromUri($url);
+            $this->updater = $repoConfig['tuf']['_updater'] ?? new ComposerCompatibleUpdater($fetcher, [], new FileStorage($repoPath));
```

Comment:
I know we are already doing this for `_fileFetcher` but it seems weird to accept config options that should only ever be used in testing.  I don't know enough about how Composer plugins work but it makes me wonder if another plugin would ever have a chance to alter this config to say pass malicious file fetcher or updater in these keys.

It seems like if we don't want `_updater` to be used during a non-test run we should do whatever we can to make that impossible.

Looking at the test 
```
return $this->composer->getRepositoryManager()
            ->createRepository('composer', $config);
```

Because `TufValidatedComposerRepository` is not final could we extend it as `TestTufValidatedComposerRepository` to be able to set the fetcher and updater in some way then use `\Composer\Repository\RepositoryManager::setRepositoryClass` to make it use our test class. The maybe avoiding having this test only code in our class?

Reply:
I proposed an alternate approach in #22, although I don't know if it's really much of an improvement.

A lot of these problems stem from the fact that ComposerRepository::$config is private, and I need to be able to modify the configuration in the constructor _before_ it's given to the parent class. If that property were protected, this would be a non-issue, and I could move all of this logic into a different method, rather than being constrained by the fact that it's all got to happen in the constructor.